### PR TITLE
Fix https://github.com/tensorflow/tensorflow/issues/8207

### PR DESCRIPTION
**Pull request by aodhan-domhnaill**
@martinwicke

### CODE REVIEW

File: tensorflow/core/ops/math_ops.cc [hunk 1]
```diff
@@ -1561,8 +1561,8 @@ If the sum is empty for a given segment ID `i`, `output[i] = 0`.
 
 `num_segments` should equal the number of distinct segment IDs.
 
-<div style="width:70%; margin:auto; margin-bottom:10px; margin-top:20px;">
-<img style="width:100%" src="https://www.tensorflow.org/images/UnsortedSegmentSum.png" alt>
+<div style='width:70%; margin:auto; margin-bottom:10px; margin-top:20px;'>
+<img style='width:100%' src='../../images/UnsortedSegmentSum.png' alt>
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: @wolffg for certainly, but I think we want the absolute paths.

File: tensorflow/core/ops/array_ops.cc [hunk 2]
```diff
@@ -1353,8 +1353,8 @@ this operation will permute `params` accordingly.
 out-of-bound indices result in safe but unspecified behavior, which may include
 raising an error.
 
-<div style="width:70%; margin:auto; margin-bottom:10px; margin-top:20px;">
-<img style="width:100%" src="../../../images/Gather.png" alt>
+<div style='width:70%; margin:auto; margin-bottom:10px; margin-top:20px;'>
+<img style='width:100%' src='../../images/Gather.png' alt>
```
Comment (martinwicke) [inline tensorflow/core/ops/array_ops.cc]: I'm not sure this works -- can you link to absolute path?

@wolffg @MarkDaoust FYI -- is this part of the docs docs? I don't know what the latest guidance is.

File: tensorflow/core/ops/array_ops.cc [hunk 3]
```diff
@@ -4779,8 +4779,8 @@ The simplest form of scatter is to insert individual elements in a tensor by
 index. For example, say we want to insert 4 scattered elements in a rank-1
 tensor with 8 elements.
 
-<div style="width:70%; margin:auto; margin-bottom:10px; margin-top:20px;">
-<img style="width:100%" src="../../images/ScatterNd1.png" alt>
+<div style='width:70%; margin:auto; margin-bottom:10px; margin-top:20px;'>
+<img style='width:100%' src='../../images/ScatterNd1.png' alt>
```
Comment (martinwicke) [inline tensorflow/core/ops/array_ops.cc]: I'll omit comment on the links for now -- let's hear where we end up on those.

File: tensorflow/core/ops/image_ops.cc [hunk 4]
```diff
@@ -358,9 +358,9 @@ try_recover_truncated:  If true try to recover an image from truncated input.
 acceptable_fraction: The minimum required fraction of lines before a truncated
   input is accepted.
 dct_method: string specifying a hint about the algorithm used for
-  decompression.  Defaults to "" which maps to a system-specific
-  default.  Currently valid values are ["INTEGER_FAST",
-  "INTEGER_ACCURATE"].  The hint may be ignored (e.g., the internal
+  decompression.  Defaults to '' which maps to a system-specific
+  default.  Currently valid values are ['INTEGER_FAST',
+  'INTEGER_ACCURATE'].  The hint may be ignored (e.g., the internal
```
Comment (martinwicke) [inline tensorflow/core/ops/image_ops.cc]: For these three lines, it looks to me like the string constants should be additionally enclosed in backticks (to make it look like this: `''` etc.).

File: tensorflow/core/ops/io_ops.cc [hunk 5]
```diff
@@ -84,7 +84,7 @@ REGISTER_OP("SaveV2")
 Saves tensors in V2 checkpoint format.
 
 By default, saves the named tensors in full.  If the caller wishes to save
-specific slices of full tensors, "shape_and_slices" should be non-empty strings
+specific slices of full tensors, 'shape_and_slices' should be non-empty strings
```
Comment (martinwicke) [inline tensorflow/core/ops/io_ops.cc]: these should be backticks.

File: tensorflow/core/ops/io_ops.cc [hunk 6]
```diff
@@ -115,14 +115,14 @@ Restores tensors from a V2 checkpoint.
 
 For backward compatibility with the V1 format, this Op currently allows
 restoring from a V1 checkpoint as well:
-  - This Op first attempts to find the V2 index file pointed to by "prefix", and
+  - This Op first attempts to find the V2 index file pointed to by 'prefix', and
```
Comment (martinwicke) [inline tensorflow/core/ops/io_ops.cc]: backticks

File: tensorflow/core/ops/io_ops.cc [hunk 7]
```diff
@@ -115,14 +115,14 @@ Restores tensors from a V2 checkpoint.
...
     if found proceed to read it as a V2 checkpoint;
   - Otherwise the V1 read path is invoked.
 Relying on this behavior is not recommended, as the ability to fall back to read
 V1 might be deprecated and eventually removed.
 
 By default, restores the named tensors in full.  If the caller wishes to restore
-specific slices of stored tensors, "shape_and_slices" should be non-empty
+specific slices of stored tensors, 'shape_and_slices' should be non-empty
```
Comment (martinwicke) [inline tensorflow/core/ops/io_ops.cc]: backticks

File: tensorflow/core/ops/math_ops.cc [hunk 8]
```diff
@@ -1039,18 +1039,18 @@ REGISTER_OP("MatMul")
     .Attr("T: {half, float, double, int32, complex64, complex128}")
     .SetShapeFn(shape_inference::MatMulShape)
     .Doc(R"doc(
-Multiply the matrix "a" by the matrix "b".
+Multiply the matrix 'a' by the matrix 'b'.
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: if a and b are parameter names, they should be in backticks.

File: tensorflow/core/ops/math_ops.cc [hunk 9]
```diff
@@ -1039,18 +1039,18 @@ REGISTER_OP("MatMul")
...
 
 The inputs must be two-dimensional matrices and the inner dimension of
-"a" (after being transposed if transpose_a is true) must match the
-outer dimension of "b" (after being transposed if transposed_b is
+'a' (after being transposed if transpose_a is true) must match the
+outer dimension of 'b' (after being transposed if transposed_b is
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: here too

File: tensorflow/core/ops/math_ops.cc [hunk 10]
```diff
@@ -1039,18 +1039,18 @@ REGISTER_OP("MatMul")
...
 true).
 
 *Note*: The default kernel implementation for MatMul on GPUs uses
 cublas.
 
-transpose_a: If true, "a" is transposed before multiplication.
-transpose_b: If true, "b" is transposed before multiplication.
+transpose_a: If true, 'a' is transposed before multiplication.
+transpose_b: If true, 'b' is transposed before multiplication.
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: here too

File: tensorflow/core/ops/math_ops.cc [hunk 11]
```diff
@@ -1065,11 +1065,11 @@ REGISTER_OP("SparseMatMul")
     .Attr("Tb: {float, bfloat16} = DT_FLOAT")
     .SetShapeFn(shape_inference::MatMulShape)
     .Doc(R"doc(
-Multiply matrix "a" by matrix "b".
+Multiply matrix 'a' by matrix 'b'.
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: here too

File: tensorflow/core/ops/math_ops.cc [hunk 12]
```diff
@@ -1065,11 +1065,11 @@ REGISTER_OP("SparseMatMul")
...
 
-The inputs must be two-dimensional matrices and the inner dimension of "a" must
-match the outer dimension of "b". This op is optimized for the case where at
-least one of "a" or "b" is sparse. The breakeven for using this versus a dense
+The inputs must be two-dimensional matrices and the inner dimension of 'a' must
+match the outer dimension of 'b'. This op is optimized for the case where at
+least one of 'a' or 'b' is sparse. The breakeven for using this versus a dense
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: and everywhere here

File: tensorflow/core/ops/math_ops.cc [hunk 13]
```diff
@@ -1694,13 +1694,13 @@ REGISTER_OP("SparseSegmentMeanGrad")
     .Doc(R"doc(
 Computes gradients for SparseSegmentMean.
 
-Returns tensor "output" with same shape as grad, except for dimension 0 whose
+Returns tensor 'output' with same shape as grad, except for dimension 0 whose
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: backticks

File: tensorflow/core/ops/math_ops.cc [hunk 14]
```diff
@@ -1694,13 +1694,13 @@ REGISTER_OP("SparseSegmentMeanGrad")
...
 value is output_dim0.
 
 grad: gradient propagated to the SparseSegmentMean op.
 indices: indices passed to the corresponding SparseSegmentMean op.
 segment_ids: segment_ids passed to the corresponding SparseSegmentMean op.
-output_dim0: dimension 0 of "data" passed to SparseSegmentMean op.
+output_dim0: dimension 0 of 'data' passed to SparseSegmentMean op.
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: backticks

File: tensorflow/core/ops/math_ops.cc [hunk 15]
```diff
@@ -1740,13 +1740,13 @@ REGISTER_OP("SparseSegmentSqrtNGrad")
...
 
-Returns tensor "output" with same shape as grad, except for dimension 0 whose
+Returns tensor 'output' with same shape as grad, except for dimension 0 whose
 value is output_dim0.
 
 grad: gradient propagated to the SparseSegmentSqrtN op.
 indices: indices passed to the corresponding SparseSegmentSqrtN op.
 segment_ids: segment_ids passed to the corresponding SparseSegmentSqrtN op.
-output_dim0: dimension 0 of "data" passed to SparseSegmentSqrtN op.
+output_dim0: dimension 0 of 'data' passed to SparseSegmentSqrtN op.
```
Comment (martinwicke) [inline tensorflow/core/ops/math_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 16]
```diff
@@ -186,7 +186,7 @@ v: A 1D variance Tensor with size matching the last dimension of t.
 beta: A 1D beta Tensor with size matching the last dimension of t.
   An offset to be added to the normalized tensor.
 gamma: A 1D gamma Tensor with size matching the last dimension of t.
-  If "scale_after_normalization" is true, this tensor will be multiplied
+  If 'scale_after_normalization' is true, this tensor will be multiplied
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 17]
```diff
@@ -245,7 +245,7 @@ v: A 1D variance Tensor with size matching the last dimension of t.
   This is the second output from tf.nn.moments,
   or a saved moving average thereof.
 gamma: A 1D gamma Tensor with size matching the last dimension of t.
-  If "scale_after_normalization" is true, this Tensor will be multiplied
+  If 'scale_after_normalization' is true, this Tensor will be multiplied
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 18]
```diff
@@ -471,11 +471,11 @@ the feature dimension is the third-to-last.
...
 output: 1-D with size the feature dimension of `out_backprop`.
 data_format: Specify the data format of the input and output data. With the
-    default format "NHWC", the bias tensor will be added to the last dimension
+    default format 'NHWC', the bias tensor will be added to the last dimension
     of the value tensor.
-    Alternatively, the format could be "NCHW", the data storage order of:
+    Alternatively, the format could be 'NCHW', the data storage order of:
         [batch, in_channels, in_height, in_width].
-    The tensor will be added to "in_channels", the third-to-the-last
+    The tensor will be added to 'in_channels', the third-to-the-last
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 19]
```diff
@@ -2469,7 +2469,7 @@ REGISTER_OP("QuantizedRelu")
     .Doc(R"doc(
 Computes Quantized Rectified Linear: `max(features, 0)`
 
-activations: Has the same output shape as "features".
+activations: Has the same output shape as 'features'.
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 20]
```diff
@@ -2498,7 +2498,7 @@ REGISTER_OP("QuantizedRelu6")
     .Doc(R"doc(
 Computes Quantized Rectified Linear 6: `min(max(features, 0), 6)`
 
-activations: Has the same output shape as "features".
+activations: Has the same output shape as 'features'.
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 21]
```diff
@@ -2528,7 +2528,7 @@ REGISTER_OP("QuantizedReluX")
     .Doc(R"doc(
 Computes Quantized Rectified Linear X: `min(max(features, 0), max_value)`
 
-activations: Has the same output shape as "features".
+activations: Has the same output shape as 'features'.
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/nn_ops.cc [hunk 22]
```diff
@@ -2602,7 +2602,7 @@ beta: A 1D beta Tensor with size matching the last dimension of t.
 beta_min: The value represented by the lowest quantized offset.
 beta_max: The value represented by the highest quantized offset.
 gamma: A 1D gamma Tensor with size matching the last dimension of t.
-  If "scale_after_normalization" is true, this tensor will be multiplied
+  If 'scale_after_normalization' is true, this tensor will be multiplied
```
Comment (martinwicke) [inline tensorflow/core/ops/nn_ops.cc]: backticks

File: tensorflow/core/ops/parsing_ops.cc [hunk 23]
```diff
@@ -105,7 +105,7 @@ names: A vector containing the names of the serialized protos.
   corresponding serialized protos.  These are purely useful for debugging
   purposes, and the presence of values here has no effect on the output.
   May also be an empty vector if no names are available.
-  If non-empty, this vector must be the same length as "serialized".
+  If non-empty, this vector must be the same length as 'serialized'.
```
Comment (martinwicke) [inline tensorflow/core/ops/parsing_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 24]
```diff
@@ -178,7 +178,7 @@ REGISTER_OP("Assign")
     .Doc(R"doc(
 Update 'ref' by assigning 'value' to it.
 
-This operation outputs "ref" after the assignment is done.
+This operation outputs 'ref' after the assignment is done.
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 25]
```diff
@@ -188,7 +188,7 @@ validate_shape: If true, the operation will validate that the shape
   'ref' will take on the shape of 'value'.
 use_locking: If True, the assignment will be protected by a lock;
   otherwise the behavior is undefined, but may exhibit less contention.
-output_ref:= Same as "ref".  Returned as a convenience for operations that want
+output_ref:= Same as 'ref'.  Returned as a convenience for operations that want
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 26]
```diff
@@ -202,14 +202,14 @@ REGISTER_OP("AssignAdd")
     .Doc(R"doc(
 Update 'ref' by adding 'value' to it.
 
-This operation outputs "ref" after the update is done.
+This operation outputs 'ref' after the update is done.
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 27]
```diff
@@ -202,14 +202,14 @@ REGISTER_OP("AssignAdd")
...
 This makes it easier to chain operations that need to use the reset value.
 
 ref: Should be from a `Variable` node.
 value: The value to be added to the variable.
 use_locking: If True, the addition will be protected by a lock;
   otherwise the behavior is undefined, but may exhibit less contention.
-output_ref:= Same as "ref".  Returned as a convenience for operations that want
+output_ref:= Same as 'ref'.  Returned as a convenience for operations that want
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 28]
```diff
@@ -223,14 +223,14 @@ REGISTER_OP("AssignSub")
     .Doc(R"doc(
 Update 'ref' by subtracting 'value' from it.
 
-This operation outputs "ref" after the update is done.
+This operation outputs 'ref' after the update is done.
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/state_ops.cc [hunk 29]
```diff
@@ -223,14 +223,14 @@ REGISTER_OP("AssignSub")
...
 This makes it easier to chain operations that need to use the reset value.
 
 ref: Should be from a `Variable` node.
 value: The value to be subtracted to the variable.
 use_locking: If True, the subtraction will be protected by a lock;
   otherwise the behavior is undefined, but may exhibit less contention.
-output_ref:= Same as "ref".  Returned as a convenience for operations that want
+output_ref:= Same as 'ref'.  Returned as a convenience for operations that want
```
Comment (martinwicke) [inline tensorflow/core/ops/state_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 30]
```diff
@@ -80,7 +80,7 @@ Update '*var' by subtracting 'alpha' * 'delta' from it.
 var: Should be from a Variable().
 alpha: Scaling factor. Must be a scalar.
 delta: The change.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 31]
```diff
@@ -139,7 +139,7 @@ alpha: Scaling factor. Must be a scalar.
 l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
 delta: The change.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 32]
```diff
@@ -171,7 +171,7 @@ l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var and accum.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 33]
```diff
@@ -277,7 +277,7 @@ lr: Scaling factor. Must be a scalar.
 rho: Decay factor. Must be a scalar.
 epsilon: Constant factor. Must be a scalar.
 grad: The gradient.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 34]
```diff
@@ -307,7 +307,7 @@ rho: Decay factor. Must be a scalar.
 epsilon: Constant factor. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var and accum.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 35]
```diff
@@ -406,7 +406,7 @@ var: Should be from a Variable().
 accum: Should be from a Variable().
 lr: Scaling factor. Must be a scalar.
 grad: The gradient.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 36]
```diff
@@ -477,7 +477,7 @@ grad: The gradient.
 lr: Scaling factor. Must be a scalar.
 l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 37]
```diff
@@ -535,7 +535,7 @@ accum: Should be from a Variable().
 lr: Learning rate. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var and accum.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 38]
```diff
@@ -616,7 +616,7 @@ lr: Scaling factor. Must be a scalar.
 l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
 global_step: Training step number. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 39]
```diff
@@ -650,7 +650,7 @@ lr: Learning rate. Must be a scalar.
...
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 40]
```diff
@@ -686,7 +686,7 @@ l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var and accum.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 41]
```diff
@@ -836,7 +836,7 @@ lr: Scaling factor. Must be a scalar.
 l1: L1 regulariation. Must be a scalar.
 l2: L2 regulariation. Must be a scalar.
 lr_power: Scaling factor. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: bacticks

File: tensorflow/core/ops/training_ops.cc [hunk 42]
```diff
@@ -878,7 +878,7 @@ lr: Scaling factor. Must be a scalar.
 l1: L1 regularization. Must be a scalar.
 l2: L2 regularization. Must be a scalar.
 lr_power: Scaling factor. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 43]
```diff
@@ -1000,7 +1000,7 @@ accum: Should be from a Variable().
 lr: Scaling factor. Must be a scalar.
 grad: The gradient.
 momentum: Momentum. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 44]
```diff
@@ -1039,7 +1039,7 @@ lr: Learning rate. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var and accum.
 momentum: Momentum. Must be a scalar.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 45]
```diff
@@ -1171,7 +1171,7 @@ beta1: Momentum factor. Must be a scalar.
 beta2: Momentum factor. Must be a scalar.
 epsilon: Ridge term. Must be a scalar.
 grad: The gradient.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 46]
```diff
@@ -1286,7 +1286,7 @@ lr: Scaling factor. Must be a scalar.
 epsilon: Ridge term. Must be a scalar.
 rho: Decay rate. Must be a scalar.
 grad: The gradient.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 47]
```diff
@@ -1337,7 +1337,7 @@ lr: Scaling factor. Must be a scalar.
...
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 48]
```diff
@@ -1381,7 +1381,7 @@ epsilon: Ridge term. Must be a scalar.
 rho: Decay rate. Must be a scalar.
 grad: The gradient.
 indices: A vector of indices into the first dimension of var, ms and mom.
-out: Same as "var".
+out: Same as 'var'.
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

File: tensorflow/core/ops/training_ops.cc [hunk 49]
```diff
@@ -1433,7 +1433,7 @@ epsilon: Ridge term. Must be a scalar.
...
```
Comment (martinwicke) [inline tensorflow/core/ops/training_ops.cc]: backticks

**Review by martinwicke — COMMENTED** [review]
I left lots of comments. Let's use this to fix some issues with wrong formatting in the docs, since you're touching it anyway.

**Comment by dr4b** [comment]
Sorry, I'm looking at this a little bit out of context.  Some of the places seem to require backticks, some should possibly have single quotes, but I think there are still a bunch where double quotes are proper.  Why did *all* of them have to be changed?

**Comment by aidan-plenert-macdonald** [comment]
@dr4b  My original motivation for changing them was to make the doc parseable by Protobuf. Unescaped double quotes would mess up the parser. The parser also has the problem that you can't handle quoted newlines. I have a pull request I am working on for that.

Currently the above hasn't been a problem because the docs aren't exposed in the protobuf, but removing a single line can expose them. This can prove beneficial for certain applications.

**Comment by vrv** [comment]
@MarkDaoust let me know what the next steps you suggest might be!

File: tensorflow/core/ops/array_ops.cc (hunk 2 above)
Comment (wolffg) [inline tensorflow/core/ops/array_ops.cc]: Agreed---please use absolute paths here.

File: tensorflow/core/ops/array_ops.cc (hunk 3 above)
Comment (wolffg) [inline tensorflow/core/ops/array_ops.cc]: As above---please use absolute (www.tensorflow.org/images/ScatterNd1.png)

File: tensorflow/core/ops/nn_ops.cc [hunk 50]
```diff
@@ -462,7 +462,7 @@ REGISTER_OP("BiasAddGrad")
     .Output("output: T")
     .SetShapeFn(shape_inference::BiasAddGradShape)
     .Doc(R"doc(
-The backward operation for "BiasAdd" on the "bias" tensor.
+The backward operation for 'BiasAdd' on the 'bias' tensor.
```
Comment (MarkDaoust) [inline tensorflow/core/ops/nn_ops.cc]: These need backticks as well.

**Review by MarkDaoust — COMMENTED** [review]
wolffg resolved the image link questions. 

I don't see much to add after existing comments.

**Comment by rmlarsen** [comment]
@aidan-plenert-macdonald please rebase to resolve conflicts.

**Comment by martinwicke** [comment]
I'll close this, as it appears inactive. Please create a new PR if you would like to pick this back up.

**Pull request closed without merging**